app/routes/dat.py

```
@@ -849,45 +889,50 @@ async def _match_single_file(file_path: str) -> dict:
+    definitive non-match and the caller must NOT fall back to a file-level
+    SHA1 of the container (which is meaningless against the DAT and would
+    re-read the whole file). When ``had_candidates`` is False the tool has no
+    embedded hash and the file-level fallback is the correct next step.
+    """
+    try:
+        candidates = await tool.embedded_hashes(file_path, cancel_event=cancel_event)
+    except EmbeddedHashUnavailable:
+        # Transient "couldn't derive the hash" — let the caller decide it's a
+        # non-cacheable error rather than falling back to a file-level hash.
+        raise
+    except Exception:  # pragma: no cover - best-effort fast path
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)